packages/roosterjs-plugin-picker/lib/PickerPlugin.ts

```diff
@@ -414,6 +438,46 @@ export default class PickerPlugin<T extends PickerDataProvider = PickerDataProvi
         }
     }
 
+    private onAndroidInputEvent(event: PluginInputEvent) {
+        const wordBeforCursor = this.getInlineElementBeforeCursor(event);
+        this.newInputLength = wordBeforCursor ? wordBeforCursor.length : 0;
+
+        if (
+            this.newInputLength < this.currentInputLength ||
+            (event.rawEvent as any).inputType === DELETE_CONTENT_BACKWARDS_INPUT_TYPE
+        ) {
+            this.removeNode(event);
+        }
+
+        this.currentInputLength = this.newInputLength;
+    }
+
+    private removeNode(event: PluginDomEvent) {
+        const searcher = cacheGetContentSearcher(event, this.editor);
+        const nodeBeforeCursor = searcher.getInlineElementBefore()
+            ? searcher.getInlineElementBefore().getContainerNode()
+            : null;
+        const nodeId = nodeBeforeCursor ? this.getIdValue(nodeBeforeCursor) : null;
+        const nodeAfterCursor = searcher.getInlineElementAfter();
+
+        if (
+            nodeId &&
+            nodeId.indexOf(this.pickerOptions.elementIdPrefix) == 0 &&
+            (nodeAfterCursor == null || !(nodeAfterCursor instanceof PartialInlineElement))
+        ) {
+            let replacementNode = this.dataProvider.onRemove(nodeBeforeCursor, true);
+            if (replacementNode) {
+                this.replaceNode(nodeBeforeCursor, replacementNode);
+                this.editor.runAsync(() => {
+                    this.editor.select(replacementNode, PositionType.After);

```

Comment:
I think we should only do this for Android scenario.

Reply:
Sounds good